client/concordclient/src/concord_client.cpp

```diff
@@ -85,18 +92,31 @@ void ConcordClient::subscribe(const SubscribeRequest& request,
         continue;
       }
 
-      // TODO: We fill event group with data from legacy updates.
-      // This needs to change depending on how the legacy API will be implemented.
-      EventGroup eg;
-      eg.id = update->block_id;
-      for (const auto& e : update->kv_pairs) {
-        eg.events.push_back({e.second.begin(), e.second.end()});
-      }
-      std::chrono::duration time_now = std::chrono::system_clock::now().time_since_epoch();
-      eg.record_time = std::chrono::duration_cast<std::chrono::microseconds>(time_now);
-      eg.trace_context = {};
+      // TODO: Distinguish between events and event groups. Depends on TRC API.
+      bool is_event_group = false;
+      if (is_event_group) {
+        EventGroup eg;
+        eg.id = update->block_id;
```

Comment:
Maybe I am missing something, how do we have a 1:1 correlation between event_group_id and block_id?

Reply:
There no correlation. This code is needs to be reworked once the TRC change is in. Good catch but it is example code from my point of view. I didn't bother making sense of it because it has to change anyways. Sorry about that.